compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java

```diff
@@ -131,8 +131,14 @@ private void addPlatformLibraryPaths(PackageContext packageContext,
         // Add all the jar library dependencies of current package (packageId)
         Collection<PlatformLibrary> otherJarDependencies = jBalBackend.platformLibraryDependencies(
                 packageContext.packageId(), scope);
+
+        List<String> fileNames = new ArrayList();
+        libraryPaths.stream().forEach(e -> fileNames.add(e.path().toFile().getName()));
```

Comment:
AFAIU, this filter will not work as intended. The requirement is to stop adding a jar if a path with the same filename already exists in the jarList.

Reply:
This seems to be correct. Lets test and verify